patroni/postgresql.py

```diff
@@ -1294,7 +1294,9 @@ def pick_synchronous_standby(self, cluster):
                      FROM pg_stat_replication
                     ORDER BY flush_location DESC"""):
             member = members.get(app_name)
-            if state != 'streaming' or not member or member.tags.get('nosync', False):
+            if (state != 'streaming' or not member
+                    or member.tags.get('nosync', False)
+                    or member.tags.get('nofailover', False)):
```

Comment:
> Also I'm wondering, is it really necessary to do this? At the end we have a nosync tag, which does exactly what we want. Replica wont became a synchronous if nosync is set in patroni.yaml.

On the other hand, in most cases people would want a synchronous replica to failover to it when the master fails and to avoid the data loss during the failover. What is the use case of running Patroni in a synchronous mode, which guarantees failing over only to the synchronous replica, and then allowing a replica that is not suitable for the failover as a synchronous one?

With multiple synchronous replicas there could be a valid use case of running a sync replica with ``nofailover``, for instance, using one of the two or more synchronous replicas for  real-time reporting. However, if we speculate how would we implement that with the ``synchronous_mode``, then the schema with a dedicated reporting sync replica would imply pinning that specific replica as a synchronous one, instead of letting  Patroni choose among the list of all replicas that don't have ``nosync`` tag set; if we would invent such a mechanism (i.e. as a dedicated ``always_sync`` tag), we would easily amend the logic proposed in this patch to allow such a replica to become synchronous even when the ``nofailover`` flag is set.